stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -249,6 +249,49 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 %end
   }
 
+  /// Accesses a contiguous subrange of the buffer's elements.
+  ///
+  /// The accessed slice uses the same indices for the same elements as the
+  /// original buffer uses. Always use the slice's `startIndex` property
+  /// instead of assuming that its indices start at a particular value.
+  ///
+  /// This example demonstrates getting a slice from a buffer of strings, finding
+  /// the index of one of the strings in the slice, and then using that index
+  /// in the original buffer.
+  ///
+%if Mutable:
+  ///     var streets = ["Adams", "Bryant", "Channing", "Douglas", "Evarts"]
+  ///     streets.withUnsafeMutableBufferPointer { buffer in
+  ///       let streetSlice = buffer[2..<buffer.endIndex]
+  ///       print(Array(streetSlice))
+  ///       // Prints "["Channing", "Douglas", "Evarts"]"
+  ///       let index = streetSlice.index(of: "Evarts")    // 4
+  ///       buffer[index!] = "Eustace"
+  ///     }
+  ///     print(streets.last!)
+  ///     // Prints "Eustace"
+%else:
+  ///     let streets = ["Adams", "Bryant", "Channing", "Douglas", "Evarts"]
+  ///     streets.withUnsafeBufferPointer { buffer in
+  ///       let streetSlice = buffer[2..<buffer.endIndex]
+  ///       print(Array(streetSlice))
+  ///       // Prints "["Channing", "Douglas", "Evarts"]"
+  ///       let index = streetSlice.index(of: "Evarts")    // 4
+  ///       print(buffer[index!])
+  ///       // Prints "Evarts"
+  ///     }
+%end
+  ///
+  /// - Note: Bounds checks for `bounds` are performed only in debug mode.
+  ///
+  /// - Parameter bounds: A range of the buffer's indices. The bounds of
+  ///   the range must be valid indices of the buffer.
+  ///
+%if Mutable:
+  /// - Complexity: O(1) for reading, O(N) for writing
```

Comment:
I see where the complexity bounds comes from, but I think it's misleading. It subtly reflects the fact that "reading" is deferred until the slice is used. And `N` will likely be mistaken for the buffer count. Since there's nothing magic or unexpected happening here, why talk about complexity? @airspeedswift, is this something we need to document?

Reply:
This is the same complexity that's documented at the protocol level, so we can remove it here. We only document on types when there's a deviation from the expectation (like `Set.count`).